project/frontend/src/components/ButtonGroup.js

```diff
@@ -9,11 +9,38 @@ import React from 'react';
  */
 function ButtonGroup(props) {
   const { labelList } = props;
-  const buttonList = [];
-  for (let delta = 0; delta < labelList.length; delta++) {
-    buttonList.push(<button>{labelList[delta]}</button>);
-  }
-  return <div className='btn-group'>{buttonList}</div>;
+  const [backgroundIsGray, setBackgroundIsGray] = useState({
+    button0: false,
+    button1: false,
+    button2: false,
+    button3: false,
+  });
+
+  const onClickButton = (props) => {
+    let tempBackgroundIsGray = Object.assign({}, backgroundIsGray);
+    tempBackgroundIsGray[props] = !tempBackgroundIsGray[props];
+    setBackgroundIsGray(tempBackgroundIsGray);
+  };
+
+  const createButtons = () => {
+    const buttonList = [];
+    for (let delta = 0; delta < labelList.length; delta++) {
```

Comment:
Sorry to be so nit about all the naming of stuff lol, but I'm not sure if delta is the best name here? I'm guessing this is based on a different PR where we used delta to describe the offset from 0, but I think here we could just use `i` or `buttonIndex` or something along those lines?